### PR TITLE
LitElement conversion part 2: src and unit tests

### LINKED ISSUE #134: LitElement conversion part 2: src and unit tests
### Config tweaks

1. Remove `npm run test:visual` from `.travis.yml`

2. Remove `theme` from `lint:js` npm script (ESLint)

### Convert src

Convert `src` files to LitElement using this structure:

- `vaadin-split-layout-css.ts` - default styles

- `vaadin-split-layout-base.ts` - base class (extend `VaadinElement`)

- `vaadin-split-layout-mixin.ts` - mixin containing component logic

- `vaadin-split-layout.ts` - custom element definition

### Convert tests

Convert unit tests in `test/unit` folder. Use the test helpers:

- `@vadin/vaadin-components-dev-dependencies/testing.js`

- `@vadin/vaadin-components-dev-dependencies/keys.js`

## Coverage

- run `es-dev-server` and go to http://localhost:8000/coverage/

- adjust coverage thresholds in `karma.conf.js` as necessary

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #134

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ addons:
 script:
 
 - if [[ "$TRAVIS_EVENT_TYPE" != "pull_request" ]]; then
-    npm run test:sauce && npm run test:visual;
+    npm run test:sauce;
```
Comment (hongduc-phan) [inline .travis.yml]: Nit: why we do not have visual tests in Lit-element version any more?

Comment (hongduc-phan) [inline .travis.yml]: why we do not need visual test in Lit version anymore? just curious about it.

File: src/vaadin-split-layout-mixin.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      const distance = isVertical ? dy : dx;
+      const isRtl = !isVertical && this.getAttribute('dir') === 'rtl';
+      const dirDistance = isRtl ? -distance : distance;
+
+      const { primary, secondary, container } = this._startSize;
+
+      this._setFlexBasis(this._primaryChild, primary + dirDistance, container);
+      this._setFlexBasis(this._secondaryChild, secondary - dirDistance, container);
+
+      if (event.detail.state === 'end') {
```
Comment (Haprog) [inline src/vaadin-split-layout-mixin.ts]: Can be shortened
```suggestion
      if (state === 'end') {
```

File: test/unit/split-layout.test.ts [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+      await nextFrame();
+      first = layout.querySelector('#first') as HTMLElement;
+      second = layout.querySelector('#second') as HTMLElement;
+    });
+
+    it('should make split-layout a flex container', () => {
+      expect(getComputedStyle(layout).display).to.equal('flex');
+    });
+
+    it('should set flex: auto on the content elements', () => {
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/split-layout.test.ts]: Done.

File: test/unit/split-layout.test.ts [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+      const getComputedFlexStyle = (el: HTMLElement) => {
+        const style = getComputedStyle(el);
+        return style.flex || [style.flexGrow, style.flexShrink, style.flexBasis].join(' ');
+      };
+
+      expect(getComputedFlexStyle(first)).to.equal('1 1 auto');
+      expect(getComputedFlexStyle(second)).to.equal('1 1 auto');
+    });
+
+    it('should set overflow: visible on the splitter', () => {
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/split-layout.test.ts]: Done.

File: test/unit/split-layout.test.ts [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+    it('should be distributed to slots', () => {
+      expect(first.assignedSlot!.getAttribute('name')).to.equal('primary');
+      expect(second.assignedSlot!.getAttribute('name')).to.equal('secondary');
+    });
+
+    it('should have a splitter between slots', () => {
+      const prev = splitter.previousElementSibling as HTMLSlotElement;
+      const next = splitter.nextElementSibling as HTMLSlotElement;
+      expect(prev.assignedNodes({ flatten: true })[0]).to.equal(first);
+      expect(next.assignedNodes({ flatten: true })[0] || next).to.equal(second);
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/split-layout.test.ts]: Removed, thanks.

File: test/unit/split-layout.test.ts [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+    });
+
+    it('should set pointer-events: none to panels on mousedown event', () => {
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/split-layout.test.ts]: Done.

File: test/unit/split-layout.test.ts [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+      first.style.pointerEvents = 'visible';
+      second.style.pointerEvents = 'visible';
+
+      mousedown(splitter);
+      mouseup(splitter);
+      expect(getComputedStyle(first).pointerEvents).to.equal('visible');
+      expect(getComputedStyle(second).pointerEvents).to.equal('visible');
+    });
+
+    it('should set pointer-events: none to panels on touchstart event', () => {
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/split-layout.test.ts]: Done.

File: src/vaadin-split-layout-mixin.ts (hunk 2 above)
Comment (web-padawan) [reply src/vaadin-split-layout-mixin.ts]: Thanks, done.

**Pull request merged**